std/internal/cstring.d

```diff
@@ -111,26 +111,25 @@ if (isSomeChar!To && (isInputRange!From || isSomeString!From) &&
         auto r = cast(const(CF)[])str;  // because inout(CF) causes problems with byUTF
         if (r is null)  // Bugzilla 14980
         {
+            res._length = 0;
             res._ptr = null;
             return res;
         }
     }
     else
         alias r = str;
-    To[] q = res._buff;
     foreach (const c; byUTF!(Unqual!To)(r))
     {
-        if (i + 1 == q.length)
+        if (i + 1 == p.length)
         {
-            p = trustedRealloc(p, i, res._buff, strLength, p_is_onstack);
+            p = trustedRealloc(p, strLength, p_is_onstack);
             p_is_onstack = false;
-            q = p;
         }
-        q[i++] = c;
+        p[i++] = c;
     }
-    q[i] = 0;
+    p[i] = 0;
     res._length = i;
-    res._ptr = p_is_onstack ? res.useStack : &p[0];
+    res._ptr = p_is_onstack ? res.useStack : p.ptr;
```

Comment:
You can't do that:

```
std/file.d(2338): Error: @safe function 'std.file.mkdirRecurse' cannot call @system function 'std.file.exists!(const(char)[]).exists'
std/file.d(2344): Error: @safe function 'std.file.mkdirRecurse' cannot call @system function 'std.file.ensureDirExists!().ensureDirExists'
std/file.d(3099): Error: @safe constructor 'std.file.DirEntry.this' cannot call @system function 'std.file.exists!string.exists'
std/file.d(2338): Error: @safe function 'std.file.mkdirRecurse' cannot call @system function 'std.file.exists!(const(char)[]).exists'
std/file.d(2344): Error: @safe function 'std.file.mkdirRecurse' cannot call @system function 'std.file.ensureDirExists!().ensureDirExists'
std/file.d(3099): Error: @safe constructor 'std.file.DirEntry.this' cannot call @system function 'std.file.exists!string.exists'
std/stdio.d(423): Error: @safe constructor 'std.stdio.File.this' cannot call @system function 'std.stdio.fopen!(string, const(char)[]).fopen'
std/stdio.d(423): Error: @safe constructor 'std.stdio.File.this' cannot call @system function 'std.stdio.fopen!(string, const(char)[]).fopen'
std/experimental/logger/filelogger.d(76): Error: @safe constructor 'std.experimental.logger.filelogger.FileLogger.this' cannot call @system function 'std.file.exists!string.exists'
std/experimental/logger/filelogger.d(76): Error: @safe constructor 'std.experimental.logger.filelogger.FileLogger.this' cannot call @system function 'std.file.exists!string.exists'
```

Reply:
Yep, plus a second `To[]` variable (now `heapBuffer` instead of `q`) is required for safe-ty.